core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -1012,7 +961,8 @@ public static class ShareSessionCache {
         private Map<ShareSessionKey, ShareSession> sessions = new HashMap<>();
 
         // Maps last used times to sessions.
-        private TreeMap<LastUsedKey, ShareSession> lastUsed = new TreeMap<>();
+        // Visible for testing
+        public TreeMap<LastUsedKey, ShareSession> lastUsed = new TreeMap<>();
```

Comment:
nit: Maybe instead of changing access to public, we can have get function to return these values for testing. Same for line 439 and 441 above.

Reply:
Yep, that's better. I have included the changes in the latest commit